portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/contentsecuritypolicy/ContentSecurityPolicy.testcase

```diff
@@ -0,0 +1,80 @@
+@component-name = "portal-security"
+definition {
+
+	property ci.retries.disabled = "true";
+	property portal.release = "true";
+	property portal.upstream = "true";
+	property testray.main.component.name = "Content Security Policy";
+
+	setUp {
+		TestCase.setUpPortalInstance();
+
+		User.firstLoginPG();
+	}
+
+	tearDown {
+		var testPortalInstance = PropsUtil.get("test.portal.instance");
+
+		if (${testPortalInstance} == "true") {
+			PortalInstances.tearDownCP();
+		}
+		else {
+			PortalSettings.tearDownAuthenticationCP();
+
+			PagesAdmin.tearDownCP();
+		}
+	}
+
+	@description = "This is a use case for LPS-182349. TC-3."
+	@priority = 4
+	test VerifyCSPCouldConfiguredByWebId {
+		property test.name.skip.portal.instance = "ContentSecurityPolicy#VerifyCSPCouldConfiguredByWebId";
+
+		task ("Add the OSGI configuration file") {
+			ContentSecurityPolicy.deployInstanceConfigFile(osgiConfigFileName = "com.liferay.portal.security.content.security.policy.configuration.ContentSecurityPolicyConfiguration.scoped~custom-image-source-webid.config");
+		}
+
+		ContentSecurityPolicy.goToContentSecurityPolicyInstanceSettings();
+
+		AssertTextPresent(
+			locator1 = "ContentSecurityPolicy#CONTENT_SECURITY_POLICY_TEXT",
+			value1 = "img-src domain.liferay.com");
+
+		var baseURL = PropsUtil.get("portal.url");
+
+		var curl = '''${baseURL} -X HEAD -I''';
+
+		var response = JSONCurlUtil.post(${curl});
+
+		TestUtils.assertPartialEquals(
+			actual = ${response},
+			expected = "Content-Security-Policy: img-src domain.liferay.com");
```

Comment:
Line 37-Line51: Half of the steps are placed in the task, and the other half are not placed in the task, which looks strange. The same issue is in VerifyCSPCouldConfiguredToBlockImageSource.